services/app-api/src/resolvers/contract/updateContractDraftRevision.ts

```diff
@@ -0,0 +1,143 @@
+import { ForbiddenError, UserInputError } from 'apollo-server-lambda'
+import { isStateUser } from '../../domain-models'
+import type { MutationResolvers } from '../../gen/gqlServer'
+import { logError, logSuccess } from '../../logger'
+import type { Store } from '../../postgres'
+import { NotFoundError } from '../../postgres'
+import {
+    setErrorAttributesOnActiveSpan,
+    setResolverDetailsOnActiveSpan,
+    setSuccessAttributesOnActiveSpan,
+} from '../attributeHelper'
+import type { LDService } from '../../launchDarkly/launchDarkly'
+import { GraphQLError } from 'graphql/index'
+import { validateContractDraftRevisionInput } from '../../domain-models/contractAndRates'
+
+export function updateContractDraftRevision(
+    store: Store,
+    launchDarkly: LDService
+): MutationResolvers['updateContractDraftRevision'] {
+    return async (_parent, { input }, context) => {
+        const { user, ctx, tracer } = context
+        const span = tracer?.startSpan('updateContractDraftRevision', {}, ctx)
+        setResolverDetailsOnActiveSpan(
+            'updateContractDraftRevision',
+            user,
+            span
+        )
+        const featureFlags = await launchDarkly.allFlags(context)
+        const { draftRevision, contractID } = input
+
+        if (draftRevision.submitInfo) {
```

Comment:
why does this input have submitInfo? I would expect only the FormData to be passed in as an argument to this mutation

Reply:
Another thing I thought of: this is user submitted data, so we cannot rely on submitInfo being honest. We correctly error out lower down based on the fetched contract, that's the right way to validate this. Even checking this from the user input is a bad idea because someone might accidentally think it's trustworthy later on. Better to not let users send data that we can't trust.